core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3058,8 +3059,13 @@ private[spark] object Utils
    */
   lazy val isG1GC: Boolean = {
     Try {
-      ManagementFactory.getGarbageCollectorMXBeans.asScala
-        .exists(_.getName.contains("G1"))
+      // SPARK-48505: If the initialization probe of `HotSpotDiagnosticMXBean` is successful,
+      // the reflection API can be avoided in subsequent operations.
+      val clazz = Utils.classForName("com.sun.management.HotSpotDiagnosticMXBean")
+      val hotSpotDiagnosticMXBean = ManagementFactory.getPlatformMXBean(clazz)
+        .asInstanceOf[HotSpotDiagnosticMXBean]
+      val useG1GC = hotSpotDiagnosticMXBean.getVMOption("UseG1GC").getValue
+      "true".equals(useG1GC)
```

Comment:
cc @rednaxelafx FYI

Reply:
Have I understood your suggestion correctly? But it seems like the detection for class initialization is not necessary here as it's already wrapped in a `Try`?